packages/amplify-provider-awscloudformation/src/transform-graphql-schema.ts

```diff
@@ -454,8 +454,8 @@ export async function transformGraphQLSchema(context, options) {
   }
 
   // for the predictions directive get storage config
-  const s3Resource = s3ResourceAlreadyExists(context);
-  const storageConfig = s3Resource ? getBucketName(context, s3Resource, backEndDir) : undefined;
+  const s3ResourceName = await invokeS3GetResourceName(context);
+  const storageConfig = s3ResourceName ? await getBucketName(context, s3ResourceName ) : undefined;
```

Comment:
I have also done these change in my gql PR, I can take up your changes since this uses S3 category APIs